### ISSUE TITLE
Error #255

### ISSUE THREAD

**Issue by estherflorin** [openmeeg/openmeeg#74]
Hello, 
I am creating a head model with brainstorm using opnemeeg. It is running on a windows 7 pc with Matlab R2014b (128 GB RAM and 1.4 TB free disk space) . When trying to create the head model (about 1922 vertices for the layers) using the adaptive integration, I am getting the error message below. What could be the problem?
Best wishes,
Esther 

---

*\* Error: OpenMEEG call: om_gain -EEG
*\*       "C:\Users\florine\Documents\brainstorm_db\resting_young\anat\S005\tess_innerskull_bem_1922V_openmeeg.bin"
*\*       "C:\Users\florine\Documents\brainstorm_tmp\openmeeg_dsm.bin"
*\*       "C:\Users\florine\Documents\brainstorm_tmp\openmeeg_h2em.bin"
*\*       "C:\Users\florine\Documents\brainstorm_tmp\openmeeg_gain_eeg.bin"
*\* OpenMEEG error #255: 
*\* om_gain version 2.2.0 (exported) compiled at Feb 19 2013 14:00:34
*\* 
*\* | ------ om_gain
*\* | -EEG
*\* | C:\Users\florine\Documents\brainstorm_db\resting_young\anat\S005\tess_innerskull_bem_1922V_openmeeg.bin
*\* | C:\Users\florine\Documents\brainstorm_tmp\openmeeg_dsm.bin
*\* | C:\Users\florine\Documents\brainstorm_tmp\openmeeg_h2em.bin
*\* | C:\Users\florine\Documents\brainstorm_tmp\openmeeg_gain_eeg.bin
*\* | -----------------------
*\* Exception: Bad storage type in file C:\Users\florine\Documents\brainstorm_tmp\openmeeg_dsm.bin. Doing my best....
*\* 
*\* Many OpenMEEG crashes are due to a lack of memory.
*\* Reduce the number of vertices on each BEM layer and try again.

---

**Comment by agramfort** [comment]
hi Esther,

does it work if you reduce the number of vertices?

Alex

**Comment by estherflorin** [comment]
Hi Alex,
I Trier with 1000 and it still did n�tig work.
Esther
________________________________________
Von: Alexandre Gramfort [notifications@github.com]
Gesendet: Donnerstag, 14. Mai 2015 22:11
An: openmeeg/openmeeg
Cc: Esther Florin
Betreff: Re: [openmeeg] Error #255 (#74)

hi Esther,

does it work if you reduce the number of vertices?

Alex

�

**Comment by agramfort** [comment]
>
> I Trier with 1000 and it still did n�tig work.
>

:( are you working on a new machine with a fresh install of OpenMEEG? did
it work in the past?

**Comment by estherflorin** [comment]
Yes, it is a new machine and therefore a new install. I also tried a reinstall at the beginning of the week. I had not tried it before. So I cannot say, if it worked before.
Esther
________________________________________
Von: Alexandre Gramfort [notifications@github.com]
Gesendet: Freitag, 15. Mai 2015 11:44
An: openmeeg/openmeeg
Cc: Esther Florin
Betreff: Re: [openmeeg] Error #255 (#74)

>
> I tried with 1000 and it still did not work.
>

:( are you working on a new machine with a fresh install of OpenMEEG? did
it work in the past?

�

**Comment by papadop** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

On 05/15/2015 11:53 AM, estherflorin wrote:
> Yes, it is a new machine and therefore a new install. I also tried
> a reinstall at the beginning of the week. I had not tried it
> before. So I cannot say, if it worked before. Esther

The error message has nothing to do with storage size, it is storage
type that is wrong. Can you check the characteristics of the file

C:\Users\florine\Documents\brainstorm_tmp\openmeeg_dsm.bin

I suspect it is a matlab file (but it can also be a bin or an ascii
file), but is it a matrix, a sparse matrix, a vector or a structure
(in which we store symmetric matrices).

If it's a .bin file, there are tools to read them with matlab and
check its properties. Or you can put it somewhere where we can have a
look (preferably create a repo with all the OpenMEEG files, so that we
can reproduce the problem).

Also, there was a mistake in the OpenMEEG brainstorm generation that
we identified recently in february, but it looks different. Maybe
there is a remaining similar problem. I suppose you have the latest
brinasorm version.


     I hope that we can solve the problem rapidly.

	Theo.
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2

iEYEARECAAYFAlVVxgQACgkQEr8WrU8nPV1E5ACgp+q+R7QkyEEIP+pLQXsrSuHR
FkIAoLvhb6D1ivzaV+3hSmk4QlVJOL6m
=I6Eq
-----END PGP SIGNATURE-----

**Comment by estherflorin** [comment]
Hello Theo,
Matlab did not like to read the file. It seems to be a text file, which is about 4 GB large. I have placed all files within my brainstorm_tmp folder in  a shared folder in dropbox. Here is the link
https://www.dropbox.com/sh/u24drdr5cfj94az/AABnS_LeRK7jVUEH8ALdxvnYa?dl=0
The brainstorm version is from May 1st. However, I do get the same error with todays version of brainstorm. Let me know, if you need further information.
Best wishes,
Esther


Von: Theodore Papadopoulo [mailto:notifications@github.com]
Gesendet: Freitag, 15. Mai 2015 12:10
An: openmeeg/openmeeg
Cc: Esther Florin
Betreff: Re: [openmeeg] Error #255 (#74)

-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

On 05/15/2015 11:53 AM, estherflorin wrote:
> Yes, it is a new machine and therefore a new install. I also tried
> a reinstall at the beginning of the week. I had not tried it
> before. So I cannot say, if it worked before. Esther

The error message has nothing to do with storage size, it is storage
type that is wrong. Can you check the characteristics of the file

C:\Users\florine\Documents\brainstorm_tmp\openmeeg_dsm.bin

I suspect it is a matlab file (but it can also be a bin or an ascii
file), but is it a matrix, a sparse matrix, a vector or a structure
(in which we store symmetric matrices).

If it's a .bin file, there are tools to read them with matlab and
check its properties. Or you can put it somewhere where we can have a
look (preferably create a repo with all the OpenMEEG files, so that we
can reproduce the problem).

Also, there was a mistake in the OpenMEEG brainstorm generation that
we identified recently in february, but it looks different. Maybe
there is a remaining similar problem. I suppose you have the latest
brinasorm version.


I hope that we can solve the problem rapidly.

Theo.
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2

iEYEARECAAYFAlVVxgQACgkQEr8WrU8nPV1E5ACgp+q+R7QkyEEIP+pLQXsrSuHR
FkIAoLvhb6D1ivzaV+3hSmk4QlVJOL6m
=I6Eq
-----END PGP SIGNATURE-----

**Comment by papadop** [comment]
Oh it is a .bin file which I would like to remove for 5 years now as it is unsafe. But if the problem is what I tend to think, we might try to improve the error message...

First let's remove the obvious, I suppose you are using Windows 7 64 bits version or otherwise your memory of 128Gb is useless, right ? This is very very very probably the case, or otherwise you could not have generated such a big file in the first place...

Then, assuming it is indeed a 64 bits windows installation, can you try the following command (om_matrix_info should come with your openmeeg installation):

om_matrix_info -i openmeeg_dsm.bin

For me the output is:

Loading : /user/papadop/home/Downloads/OM/openmeeg_dsm.bin
Dimensions : 13446 x 45006
Min Value : -119.536 (1205,20350)
Max Value : 129.547 (1582,20393)
First Values
-0.0457396 0.0112958 0.00428926 -0.0574582 0.0141451 0.00174376 -0.0360772 0.0116272 0.00773741 -0.0832005 
-0.0967332 0.0190873 -0.00161718 -0.119975 0.0228516 -0.00913885 -0.0731525 0.0215642 0.00775376 -0.200726 
-0.17854 0.0270657 0.0603089 -0.252009 0.0445101 0.0742415 -0.126811 0.0311994 0.0632287 -0.392592 
-0.135434 0.0717627 0.0069868 -0.154782 0.0816916 -0.00713867 -0.098044 0.0654664 0.0166602 -0.279484 
-0.397675 0.211344 0.153668 -0.552172 0.331427 0.163449 -0.239815 0.164089 0.133738 -1.25296 
-0.765775 0.0399272 0.524353 -1.38894 0.120349 1.04905 -0.439072 0.0820721 0.411545 -1.84891 
-0.120863 0.0323485 0.000719132 -0.123489 0.0353028 -0.00510557 -0.0949945 0.0373003 0.00652231 -0.202002 
-0.491194 0.230372 0.0661816 -0.532089 0.284032 0.0171251 -0.310181 0.192241 0.0811538 -1.20271 
-4.30502 3.60888 3.90824 -3.63725 12.3539 8.22731 -1.45641 1.3536 1.46345 38.8333 
-4.07768 -1.34539 2.83233 -6.12582 -8.31902 8.39007 -2.31958 -0.435999 2.37925 -2.45299 

Which proves that my linux 64bits machine is capable of reading the file (so it is not corrupted)....
If your installation also does, then I have to dive into the problem deeper (but for that I miss the file tess_innerskull_bem_1922V_openmeeg.bin). And you can forget the rest of this comment.

If you get an error from this command: 
- either your version of openmeeg is not 64bits (but that would be strange given the big file has been created. What is the name of the file in  .brainstorm/openmeeg ? This might give a hint). If you have a 32 bits openmeeg, then  openmeeg is limited to file of (slightly less than) 4Gb. You might also use the information on http://superuser.com/questions/358434/how-to-check-if-a-binary-is-32-or-64-bit-on-windows to check whether the om_gain binary is 32 or 64 bits (I'll try to do so on the packages we have generated later).
- or there is a problem with the compiler or the platform which does not deal correctly with 64bits quantities, and I have to investigate on the virtual windows machine we used to generate the package (which will be a little bit difficult given the size of the files). This is probably the proper case or otherwise I cannot understand how you could have generated such a big file...

Hope this helps a little...

**Comment by estherflorin** [comment]
Hello Theo,
Yes, I am using Windows 7 64 bit.
I am not able to read the file openmeeg_dsm.bin on my machine . Here is the error message:

Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation. Alle Rechte vorbehalten.

C:\Users\florine\.brainstorm\openmeeg\win64>om_matrix_info -i openmeeg_dsm.bin
om_matrix_info version 2.2.0 (exported) compiled at Feb 19 2013 13:55:21

Loading : openmeeg_dsm.bin
Exception: Unable to open the file openmeeg_dsm.bin for reading Doing my best...
.

C:\Users\florine\.brainstorm\openmeeg\win64>



Then I checked, if om_gain.exe is 64 bit and this seems to be the case:
>> !sigcheck om_gain.exe
Sigcheck v2.20 - File version and signature viewer
Copyright (C) 2004-2015 Mark Russinovich
Sysinternals - www.sysinternals.com

C:\Users\florine\.brainstorm\openmeeg\win64\om_gain.exe:
                Verified:              Unsigned
                Link date:            15:09 19.02.2013
                Publisher:            n/a
                Description:        n/a
                Product:               n/a
                Prod version:     n/a
                File version:        n/a
                MachineType:   64-bit
>>

So I guess the installation is 64bit. Any suggestion, what else I should check?
Thank you
Esther
Von: Theodore Papadopoulo [mailto:notifications@github.com]
Gesendet: Montag, 18. Mai 2015 20:21
An: openmeeg/openmeeg
Cc: Esther Florin
Betreff: Re: [openmeeg] Error #255 (#74)


Oh it is a .bin file which I would like to remove for 5 years now as it is unsafe. But if the problem is what I tend to think, we might try to improve the error message...

First let's remove the obvious, I suppose you are using Windows 7 64 bits version or otherwise your memory of 128Gb is useless, right ? This is very very very probably the case, or otherwise you could not have generated such a big file in the first place...

Then, assuming it is indeed a 64 bits windows installation, can you try the following command (om_matrix_info should come with your openmeeg installation):

om_matrix_info -i openmeeg_dsm.bin

For me the output is:

Loading : /user/papadop/home/Downloads/OM/openmeeg_dsm.bin
Dimensions : 13446 x 45006
Min Value : -119.536 (1205,20350)
Max Value : 129.547 (1582,20393)
First Values
-0.0457396 0.0112958 0.00428926 -0.0574582 0.0141451 0.00174376 -0.0360772 0.0116272 0.00773741 -0.0832005
-0.0967332 0.0190873 -0.00161718 -0.119975 0.0228516 -0.00913885 -0.0731525 0.0215642 0.00775376 -0.200726
-0.17854 0.0270657 0.0603089 -0.252009 0.0445101 0.0742415 -0.126811 0.0311994 0.0632287 -0.392592
-0.135434 0.0717627 0.0069868 -0.154782 0.0816916 -0.00713867 -0.098044 0.0654664 0.0166602 -0.279484
-0.397675 0.211344 0.153668 -0.552172 0.331427 0.163449 -0.239815 0.164089 0.133738 -1.25296
-0.765775 0.0399272 0.524353 -1.38894 0.120349 1.04905 -0.439072 0.0820721 0.411545 -1.84891
-0.120863 0.0323485 0.000719132 -0.123489 0.0353028 -0.00510557 -0.0949945 0.0373003 0.00652231 -0.202002
-0.491194 0.230372 0.0661816 -0.532089 0.284032 0.0171251 -0.310181 0.192241 0.0811538 -1.20271
-4.30502 3.60888 3.90824 -3.63725 12.3539 8.22731 -1.45641 1.3536 1.46345 38.8333
-4.07768 -1.34539 2.83233 -6.12582 -8.31902 8.39007 -2.31958 -0.435999 2.37925 -2.45299

Which proves that my linux 64bits machine is capable of reading the file (so it is not corrupted)....
If your installation also does, then I have to dive into the problem deeper (but for that I miss the file tess_innerskull_bem_1922V_openmeeg.bin). And you can forget the rest of this comment.

If you get an error from this command:

  *   either your version of openmeeg is not 64bits (but that would be strange given the big file has been created. What is the name of the file in .brainstorm/openmeeg ? This might give a hint). If you have a 32 bits openmeeg, then openmeeg is limited to file of (slightly less than) 4Gb. You might also use the information on http://superuser.com/questions/358434/how-to-check-if-a-binary-is-32-or-64-bit-on-windows to check whether the om_gain binary is 32 or 64 bits (I'll try to do so on the packages we have generated later).
  *   or there is a problem with the compiler or the platform which does not deal correctly with 64bits quantities, and I have to investigate on the virtual windows machine we used to generate the package (which will be a little bit difficult given the size of the files). This is probably the proper case or otherwise I cannot understand how you could have generated such a big file...

Hope this helps a little...

**Comment by papadop** [comment]
OK. So I ave now an easy way to reproduce the problem with the om_matrix command. From the win64 name, it clearly seems that Brainstorm installed out 64 bits package. Maybe it is a build problem on our side (we let tools select the 64 bit mode for us, and maybe we should not be confident here). Give me a few days to check on our windows machines....

**Comment by estherflorin** [comment]
Ok. Thank you.
Esther

Von: Theodore Papadopoulo [mailto:notifications@github.com]
Gesendet: Dienstag, 19. Mai 2015 11:12
An: openmeeg/openmeeg
Cc: Esther Florin
Betreff: Re: [openmeeg] Error #255 (#74)


OK. So I ave now an easy way to reproduce the problem with the om_matrix command. From the win64 name, it clearly seems that Brainstorm installed out 64 bits package. Maybe it is a build problem on our side (we let tools select the 64 bit mode for us, and maybe we should not be confident here). Give me a few days to check on our windows machines....

**Comment by papadop** [comment]
Esther, can you try the command that failed but replacing the dsm file with the one available at:
https://www.dropbox.com/s/sd4w51p578liaw7/openmeeg_dsm.mat?dl=0
If everything went well this is a matlab file corresponding to the one you gave me (this already showed some problems in a library we use in OpenMEEG on my machine that has only (!) 8Gb of memory with your file). As you can see it less than 50% of the size, which is one of the reasons I would like to see .bin files disappear. And normally it can be read directly by matlab...

I do not know whether brainstorm is able to continue if the OpenMEEG command using the matlab file succeeds... If both these conditions are true, then at least you will be no longer blocked !!!

If the command succeeds, this also strengthens my idea that there is a bug with big .bin files on windows.

**Comment by estherflorin** [comment]
Hello Theo,
So the command om_matrix_info -i openmeeg_dsm works and it is readable in matlab:
Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation. Alle Rechte vorbehalten.

C:\Users\florine>cd C:\Users/florine/.brainstorm/openmeeg/win64

C:\Users\florine\.brainstorm\openmeeg\win64>om_matrix_info -i openmeeg_dsm.mat
om_matrix_info version 2.2.0 (exported) compiled at Feb 19 2013 13:55:21

Loading : openmeeg_dsm.mat
Format : BINARY
Dimensions : 13446 x 45006
Min Value : -119.536 (1205,20350)
Max Value : 129.547 (1582,20393)
First Values
-0.0457396 0.0112958 0.00428926 -0.0574582 0.0141451 0.00174376 -0.0360772 0.011
6272 0.00773741 -0.0832005
-0.0967332 0.0190873 -0.00161718 -0.119975 0.0228516 -0.00913885 -0.0731525 0.02
15642 0.00775376 -0.200726
-0.17854 0.0270657 0.0603089 -0.252009 0.0445101 0.0742415 -0.126811 0.0311994 0
.0632287 -0.392592
-0.135434 0.0717627 0.0069868 -0.154782 0.0816916 -0.00713867 -0.098044 0.065466
4 0.0166602 -0.279484
-0.397675 0.211344 0.153668 -0.552172 0.331427 0.163449 -0.239815 0.164089 0.133
738 -1.25296
-0.765775 0.0399272 0.524353 -1.38894 0.120349 1.04905 -0.439072 0.0820721 0.411
545 -1.84891
-0.120863 0.0323485 0.000719132 -0.123489 0.0353028 -0.00510557 -0.0949945 0.037
3003 0.00652231 -0.202002
-0.491194 0.230372 0.0661816 -0.532089 0.284032 0.0171251 -0.310181 0.192241 0.0
811538 -1.20271
-4.30502 3.60888 3.90824 -3.63725 12.3539 8.22731 -1.45641 1.3536 1.46345 38.833
3
-4.07768 -1.34539 2.83233 -6.12582 -8.31902 8.39007 -2.31958 -0.435999 2.37925 -
2.45299

How should I continue now? Should I try with brainstorm now?
Thank you
Esther
_______________________________________________
Dr. rer. nat. Esther Florin
Klinik und Poliklinik für Neurologie
Universitätsklinikum Köln (AöR)
Kerpener Str. 62
50937 Köln
Germany
Tel: +49-(0)221-478-84695
esther.florin@uk-koeln.de<mailto:esther.florin@uk-koeln.de>

Von: Theodore Papadopoulo [mailto:notifications@github.com]
Gesendet: Mittwoch, 20. Mai 2015 21:04
An: openmeeg/openmeeg
Cc: Esther Florin
Betreff: Re: [openmeeg] Error #255 (#74)


Esther, can you try the command that failed but replacing the dsm file with the one available at:
https://www.dropbox.com/s/sd4w51p578liaw7/openmeeg_dsm.mat?dl=0
If everything went well this is a matlab file corresponding to the one you gave me (this already showed some problems in a library we use in OpenMEEG on my machine that has only (!) 8Gb of memory with your file). As you can see it less than 50% of the size, which is one of the reasons I would like to see .bin files disappear. And normally it can be read directly by matlab...

I do not know whether brainstorm is able to continue if the OpenMEEG command using the matlab file succeeds... If both these conditions are true, then at least you will be no longer blocked !!!

If the command succeeds, this also strengthens my idea that there is a bug with big .bin files on windows.

**Comment by papadop** [comment]
Good... My fault, I was meaning the original command...

om_gain -EEG C:\Users\florine\Documents\brainstorm_db\resting_young\anat\S005\tess_innerskull_bem_1922V_openmeeg.bin C:\Users\florine\Documents\brainstorm_tmp\openmeeg_dsm.mat C:\Users\florine\Documents\brainstorm_tmp\openmeeg_h2em.bin C:\Users\florine\Documents\brainstorm_tmp\openmeeg_gain_eeg.bin

**Comment by estherflorin** [comment]
Sorry for the slow response – I was traveling and did not have access to the computer.
From the command line it works. But how does it work with brainstorm and for the other subjects?  I assume I will need a new openmeeg_dsm.mat for each subject.
Thank you
Esther

Von: Theodore Papadopoulo [mailto:notifications@github.com]
Gesendet: Freitag, 22. Mai 2015 12:07
An: openmeeg/openmeeg
Cc: Esther Florin
Betreff: Re: [openmeeg] Error #255 (#74)


Good... My fault, I was meaning the original command...

om_gain -EEG C:\Users\florine\Documents\brainstorm_db\resting_young\anat\S005\tess_innerskull_bem_1922V_openmeeg.bin C:\Users\florine\Documents\brainstorm_tmp\openmeeg_dsm.mat C:\Users\florine\Documents\brainstorm_tmp\openmeeg_h2em.bin C:\Users\florine\Documents\brainstorm_tmp\openmeeg_gain_eeg.bin

**Comment by papadop** [comment]
It was meant as a stopgap not as a definitive solution. A solution that might work is to replace all .bin into .mat the file brainstorm3/toolbox/forward/bst_openmeeg.m and also all calls of om_load_full by just load... This would remove all uses of .bin files in brainstorm. 

Maybe we have to check with Francois (I need to find how to invite him to this discussion).

**Comment by estherflorin** [comment]
Ok. I can also get in contact with Francois, but I guess you both know better, what needs to be done.
Esther

Von: Theodore Papadopoulo [mailto:notifications@github.com]
Gesendet: Dienstag, 26. Mai 2015 10:08
An: openmeeg/openmeeg
Cc: Esther Florin
Betreff: Re: [openmeeg] Error #255 (#74)


It was meant as a stopgap not as a definitive solution. A solution that might work is to replace all .bin into .mat the file brainstorm3/toolbox/forward/bst_openmeeg.m and also all calls of om_load_full by just load... This would remove all uses of .bin files in brainstorm.

Maybe we have to check with Francois (I need to find how to invite him to this discussion).

**Comment by ftadel** [comment]
Hello,
I'm sorry for the response delay, I just came back to Montreal.

Everything is doable on the Brainstorm side. It would take me a few minutes to change all the .bin files to .mat files.  Does the current version of OpenMEEG used by Brainstorm support Matlab files?
Would OpenMEEG detect what format to use just based on the extension, or do we need additional input parameters?

Note that only the Matlab v7.3 .mat format (HDF5-based) supports files larger than 2Gb. The older formats (v6 and v7) do not work properly for larger files. Make sure to check what version of the .mat format you are using.

The packages used currently by Brainstorm are listed in bst_openmeeg.m:
http://openmeeg.gforge.inria.fr/download/OpenMEEG-2.2.0-Linux32.i386-gcc-4.3.2-static.tar.gz
http://openmeeg.gforge.inria.fr/download/OpenMEEG-2.2.0-Linux64.amd64-gcc-4.3.2-OpenMP-static.tar.gz
http://openmeeg.gforge.inria.fr/download/OpenMEEG-2.2.0-MacOSX-Intel-gcc-4.2.1-static.tar.gz
http://openmeeg.gforge.inria.fr/download/OpenMEEG-2.2.0-MacOSX-Intel-gcc-4.2.1-static.tar.gz
http://openmeeg.gforge.inria.fr/download/OpenMEEG-2.2.0-win32-x86-cl-OpenMP-shared.tar.gz
http://openmeeg.gforge.inria.fr/download/OpenMEEG-2.2.0-win64-x86_64-cl-OpenMP-shared.tar.gz

Cheers,
Francois

**Comment by papadop** [comment]
1) Does the current version of OpenMEEG used by Brainstorm support Matlab files?
Yes and I'm even more sure since Esther made tests.

2) Would OpenMEEG detect what format to use just based on the extension, or do we need additional input parameters?

Extension is enough. Actually, when the extension is not given, he will try to load the files with every format to find the proper one. The suffix is mainly used at save time to identify the proper file format. Using .bin forces the use of the .bin format.

3) Note that only the Matlab v7.3 .mat format (HDF5-based) supports files larger than 2Gb. The older formats (v6 and v7) do not work properly for larger files. Make sure to check what version of the .mat format you are using.

I'm sure we use Matlab v7.3 .mat format (HDF5-based). We force this use when we create files for this very reason (>2Gb files). Again the fact that Esther was able to make the test I asked shows that the windows version does the proper thing and this is the same for all versions. A simple test is to do the 
om_matrix_info -i openmeeg_dsm.mat (on the >4Gb file that is/was on the dropbox).

**Comment by papadop** [comment]
But we know the read the previous matlab file formats...

**Comment by papadop** [comment]
In the meantime, I have not forgotten you Esther. It's just that my windows machines have some hard time hosting the 4Gb file, and in trying to make more disk space, I broke a few things I need to restore. But I'll do my best to try to correct (if possible) the problem you encountered.

**Comment by ftadel** [comment]
I modified the bst_openmeeg.m function and replaced all the ".bin" filenames with ".mat" filenames.
I had to edit the function "om_load_full" at the end of the function, to support the reading from those .mat files (which are structures with pretty random field names...)

I did a few more adaptations in the rest of the interface to handle the "_openmeeg.mat" files that are saved in the anatomy folder (HmFile).

I haven't fully tested it, I let you try it and let me know if it works.
The update is online, just update Brainstorm (menu Help > Update Brainstorm)
